### ISSUE TITLE
Only force source-dir / dest dir to exist if relevant for operation

### ISSUE THREAD

**Issue by midgleyc** [okv/east#71]
Currently, if the source-dir directory is configured, both source-dir and dest directories are required to exist for all operations.

However, "create" shouldn't care if "dest" exists or not, and "migrate" shouldn't care whether "source-dir" exists.

My use-case is that I write the migrations in TypeScript, and want to copy only the compiled files onto production (and sometimes that I want to "east create" from a fresh repo without having run "tsc"). I get around this by having a separate `.eastrc` on prod that doesn't specify a source-dir, but it would be nicer if I didn't have to.

I can't just always run `npx east init` because that fails to make directories recursively, and my migrations directory is "dist/db/migrations", and "dist" doesn't exist.

**Comment by okv** [comment]
Hi, not a regular use-case, for sure :smiley:
I would add `-r|--recursive` flag to the `init` command, could be implemented easily since `fs.mkdir` has `recursive` option.

**Comment by midgleyc** [comment]
`recursive` came in in `10.12.0`, but if you're okay with upping the engines requirement (in `package.json`) to that I think it's a nice solution :)

**Comment by okv** [comment]
Good point about updating requirements, I think it's still a good solution considering that nodejs v10 is going to reach its end of life in the foreseeable future. I'll try to implement it soon.
